reviewrot/githubstack.py

```diff
@@ -0,0 +1,166 @@
+            # get pull requests for specified username and repo name
+            res = self.get_reviews(uname=uname, repo_name=repo_name,
+                                   state_=state_, value=value,
+                                   duration=duration)
+            response.append(res)
+        else:
+            # get all of the respositories for specified user/organization
+            repo_list = uname.get_repos()
+            if len(list(repo_list)) == 0:
+                log.debug("No repositories found for user name %s", user_name)
+            """
+            list pull requests for all of the repositories for specified
+            user/organization
+            """
+            for repo in repo_list:
+                res = self.get_reviews(uname=uname, repo_name=repo.name,
+                                       state_=state_, value=value,
+                                       duration=duration)
+                response.append(res)
+        return response
+
+    def get_reviews(self, uname, repo_name, state_=None,
+                    value=None, duration=None):
+        """
+        Fetches pull requests for specified username and repo name.
+        Formats the pull requests details and print it on console.
+
+        Args:
+            user_name (str): Github username or organization name
+            repo_name (str): Github repository name for specified
+                             username or organization
+            state_ (str): The filter(state) for pull requests, e.g, older
+                        or newer
+            value (int): The value in terms of duration for requests
+                         to be older or newer than
+            duration (str): The duration in terms of period(year,
+                            month, hour, minute) for requests to be
+                            older or newer than.
+        """
+        try:
+            # get repository object for given user/organization and repo name
+            repo = uname.get_repo(repo_name)
+        except UnknownObjectException as e:
+            log.debug('Repository %s not found for user %s',
+                      repo_name, uname.login)
+            log.debug(e)
+            raise Exception('Repository %s not found for user %s'
+                            % (repo_name, uname.login))
+        log.debug('Looking for pull requests for %s -> %s/%s ',
+                  'github', uname.login, repo_name)
+        # get list of open pull requests for a given repository
+        pull_requests = repo.get_pulls()
+        if len(list(pull_requests)) == 0:
+            log.debug('No open pull requests found for %s/%s ',
+                      uname.login, repo_name)
+        res_ = []
+        for pr in pull_requests:
+            """
+            find the relative time difference between now and
+            pull request filed to retrieve relative information
+            """
+            rel_diff = relativedelta(datetime.datetime.now(),
```

Comment:
The former can not be calculated accurately from the latter.
Timedelta object only stores seconds and microseconds, hence only minutes, hours and days could be calculated, not the month and year.